spec/lib/brexit_checker/notification_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "spec_helper"
+
+RSpec.describe BrexitChecker::Notification do
+  describe "validations" do
+    let(:notification_missing_action_id) { FactoryBot.build(:brexit_checker_notification, action_id: nil) }
+
+    it "action id can't be blank" do
+      message = "Validation failed: Action can't be blank"
+      expect { notification_missing_action_id.valid? }.to raise_error(ActiveModel::ValidationError, message)
+    end
+  end
+
+  describe "factories" do
+    it "has a valid default factory" do
+      notification = FactoryBot.build(:brexit_checker_notification)
+      expect(notification.valid?).to be(true)
+    end
+  end
+
+  describe "#action" do
+    let(:action) { FactoryBot.build(:brexit_checker_action) }
+    let(:action2) { FactoryBot.build(:brexit_checker_action) }
+    let(:notification) { FactoryBot.build(:brexit_checker_notification, action_id: action.id) }
+
+    before :each do
+      allow(BrexitChecker::Action).to receive(:load_all).and_return([action, action2])
+    end
+
+    it "returns the action relating to that notification" do
+      expect(notification.action).to eq action
+    end
+  end
+
+  describe ".load" do
```

Comment:
Great tests,
minor question/quibble on the naming.

On line 20 we're testing a method with `#action` here we're testing `.load`.
Is there something I'm missing signified by the difference in name?

Not sure if we have a GOV.UK standard for these, I've seen a lot of `#action` style descriptions around I think, but if there's not someting different between them, it'd be good to make these consistent.

Reply:
The [convention](https://rspec.rubystyle.guide/#describe-the-methods) for describing class methods is "." and for instance methods it's "#".

Reply:
ahh i see, so action is an instance.
Well TIL.. time to go back and rewrite some tests in my current project :blush: 
Thanks for the links